packages/jest-matchers/src/matchers.js

```diff
@@ -67,6 +67,18 @@ const matchers: MatchersObject = {
           `Received:\n` +
           `  ${printReceived(received)}`
       : () => {
+          if (
+            getType(received) === getType(expected) &&
+            (getType(received) === 'object' || getType(expected) === 'array') &&
+            equals(received, expected, [iterableEquality])
+          ) {
+            return (
+              matcherHint('.toBe') +
+              '\n\n' +
+              'Looks like you wanted to test for object/array equity with strict `toBe` matcher. You probably need to use `toEqual` instead.'
```

Comment:
What if we just add the new message below and leave the rest as it was? Just a new line as extra information